main.py

```diff
@@ -75,15 +91,15 @@ def train(model, iterator, optimizer, criterion, device, scheduler=None):
             
     return epoch_loss / len(iterator), correct / total
 
-def test(model, iterator, criterion, device):
-    
+def test(model, iterator, criterion, device, epoch, args):
+    global best_acc
```

Comment:
let's avoid global variables here. we can probably pass in to the test method if needed